File: apprefurbished/src/main/java/com/androidcollider/easyfin/home/HomeModule.java
package com.androidcollider.easyfin.home;

import com.androidcollider.easyfin.common.repository.Repository;

import dagger.Module;
import dagger.Provides;

/**
 * @author Ihor Bilous
 */

@Module
public class HomeModule {

    @Provides
    HomeMVP.Presenter provideHomeMVPPresenter(HomeMVP.Model model) {
        return new HomePresenter(model);
    }

    @Provides
    HomeMVP.Model provideHomeMVPModel(Repository repository) {
        return new HomeModel(repository);
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/home/HomeModule.kt
package com.androidcollider.easyfin.home

import com.androidcollider.easyfin.common.managers.chart.data.ChartDataManager
import com.androidcollider.easyfin.common.managers.format.number.NumberFormatManager
import com.androidcollider.easyfin.common.managers.rates.exchange.ExchangeManager
import com.androidcollider.easyfin.common.managers.resources.ResourcesManager
import com.androidcollider.easyfin.common.repository.Repository
import dagger.Module
import dagger.Provides

/**
 * @author Ihor Bilous
 */
@Module
class HomeModule {
    @Provides
    fun provideHomeMVPPresenter(
        model: HomeMVP.Model,
        resourcesManager: ResourcesManager,
        numberFormatManager: NumberFormatManager,
        exchangeManager: ExchangeManager,
        chartDataManager: ChartDataManager
    ): HomeMVP.Presenter {
        return HomePresenter(
            model,
            resourcesManager,
            numberFormatManager,
            exchangeManager,
            chartDataManager
        )
    }

    @Provides
    fun provideHomeMVPModel(repository: Repository): HomeMVP.Model {
        return HomeModel(repository)
    }
}